packages/glimmer-runtime/lib/compiled/opcodes/content.ts

```diff
@@ -27,12 +39,20 @@ export class AppendOpcode extends Opcode {
   evaluate(vm: VM) {
     let reference = vm.frame.getOperand();
 
-    let mapped = map(reference, normalizeTextValue);
+    let mapped = map(reference, normalizeTextOrTrustedValue);
     let cache = new ReferenceCache(mapped);
-    let node = vm.stack().appendText(cache.peek());
-
-    if (!isConst(reference)) {
-      vm.updateWith(new UpdateAppendOpcode(cache, node));
+    let value = cache.peek();
+
+    if (isSafeString(value)) {
+      let bounds = vm.stack().insertHTMLBefore(null, value);
```

Comment:
Is that something that `isConst` can determine? What if the helper is stateful and the return value changes the markup in the returned safeString? It seems like the code path would be the same.